README.md

```diff
@@ -1,2 +1,138 @@
 # elm-smart-select
 A feature rich select component written in Elm 0.19
+
+## Install
+elm install mercurymedia/elm-datetime-picker
+
+## Usage
+This package exposes two modules `SmartSelectSingle` and `SmartSelectMulti`. As their names imply, `SmartSelectSingle` can be used to pick a single element while `SmartSelectMulti` is used to select a list of elements. To keep things simple, the documentation here focuses on the `SmartSelectSingle` but both are implemented in the example app for additional reference.
+
+There are 7 steps to configure the `SmartSelect`:
+
+1. Import the SmartSelect and SmartSelect.Utilities module and expose the `SearchUnion` type. We will get back to why we need the `SearchUnion` later.
+
+```elm
+import SmartSelectSingle as SingleSelect
+import SmartSelect.Utilities exposing (SearchUnion(..))
+```
+
+2. Add the smart select to the model providing the datatype of the data to be selected and initialize it in the model init
+
+```elm
+type alias Product =
+    { id : Int
+    , name : String
+    , price : String
+    }
+
+type alias Model =
+    { ...
+    , singleSelect : SingleSelect.SmartSelect Product
+    }
+
+init : ( Model, Cmd Msg )
+init =
+    ( { ...
+      , singleSelect = SingleSelect.init
+      }
+    )
+```
+
+3. One message needs to be defined indicating that an update to the smart select has been triggered and needs to be handled. This message expects a tuple containing a `SingleSelect.Msg` and the selection state. **_Note:_** The selection state for the `SmartSelectMulti` is a List of selected entities as opposed to a Maybe entity for `SmartSelectSingle`.
+
+```elm
+type Msg
+    = ...
+    | HandleSelectUpdate ( SingleSelect.Msg Product, Maybe Product )
+```
+
+4. Configure the settings for the smart select. The `Settings` type as defined in the `SmartSelectSingle` module is below:
+
+```elm
+type alias Settings msg a =
```

Comment:
I'm not sure I agree, apart from debounceDuration, characterSearchThreshold, and closeOnSelect, which can reasonably have defaults, all of the remaining settings fields are dependent on the implementer to define.

For example, internalMsg has to come from the implementer... That is what allows us to bypass the Html.mapping. If we want to ditch that convention entirely, then we don't need to have the internalMsg at all, but otherwise, it has to be defined outside of the select component.

Furthermore, optionLabel and optionDescription also need to be defined by the implementer because the select is not aware of the structure of the data it is displaying so it would be impossible to define a function in the select that could pull the relevant field from the data.

Finally, search is so specific to an implementer's use case for both local and external searches, we couldn't realistically define a default.

So... we could provide a function that expects the problem fields I've mentioned above and returns settings with default debounce, characterThreshold, and closeOnSelect values... but does that really add much? I like that the settings are super transparent at the moment, even if they are a bit verbose.